apps/backend/charity/src/getCurrentCharity.ts

```diff
@@ -0,0 +1,51 @@
+import { Response } from "express";
+import { IDatabase } from "pg-promise";
+import { AuthenticatedRequest } from "../../lib/src/auth";
+
+export const getCurrentCharity = async (
+  req: AuthenticatedRequest,
+  res: Response,
+  db: IDatabase<object>,
+) => {
+  try {
+    const charityResult = await db.oneOrNone(
+      `SELECT c.charity_id AS id, c.name, c.description, c.start_date AS "startDate", c.end_date AS "endDate", c.image_url AS "imageUrl", 
+        jsonb_agg(jsonb_build_object('name', o.name, 'logoUrl', o.logo_url, 'donated', o.donated, 'receiving', o.receiving)) AS organizations
+       FROM charities c
+       LEFT JOIN organizations o ON c.charity_id = o.charity_id
+       WHERE NOW() BETWEEN c.start_date AND c.end_date
+       GROUP BY c.charity_id;`,
+    );
+
+    if (!charityResult) {
+      return res.status(200).json({ message: "No active charity" });
```

Comment:
Nit: is 200 the best response status, or would something else (maybe 404) be better?

Reply:
There doesn't need to be an active charity 100% of the time, not having an active charity is expected behaviour therefore 200